client/state/notification-settings/toggle-state.js

```diff
@@ -0,0 +1,64 @@
+/** @format */
+/**
+ * Internal dependencies
+ */
+import { find, get, without } from 'lodash';
+
+const toggleInStream = ( streamName, stream, setting ) => ( {
+	[ streamName ]: {
+		...stream,
+		[ setting ]: ! get( stream, setting ),
+	},
+} );
+
+const toggleInDevice = ( devices, deviceId, setting ) => {
+	const device = find( devices, { device_id: parseInt( deviceId, 10 ) } );
+	const deviceSetting = get( device, setting );
+
+	return {
+		devices: [
+			...without( devices, device ),
+			{
+				...device,
+				[ setting ]: ! deviceSetting,
+			},
+		],
+	};
+};
+
+export default {
+	wpcom( state, source, stream, setting ) {
+		return toggleInStream( 'wpcom', get( state, 'dirty.wpcom' ), setting );
+	},
+
+	other( state, source, stream, setting ) {
+		const devices = get( state, 'dirty.other.devices' );
+
+		return {
+			other: {
+				...get( state, 'dirty.other' ),
+				...( isNaN( stream )
+					? toggleInStream( stream, get( state, [ 'dirty', 'other', stream ] ), setting )
+					: toggleInDevice( devices, stream, setting ) ),
+			},
+		};
+	},
+
+	blog( state, source, stream, setting ) {
```

Comment:
Can you rename this to `blogs`? The state stores the blogs settings as `blogs`, the `getNotificationSettings` selector is always called with `blogs` parameter... I don't know how this inconsistency was introduced, but I believe it's time to fix it.